packages/core/utils/src/dml/helpers/entity-builder/build-indexes.ts

```diff
@@ -42,3 +42,33 @@ function transformWhere(where?: string): string {
 
   return where
 }
+
+/*
+  When a "belongsTo" relationship is found on the DML schema, we create a default index on the
+  foreign key name of the field.
+*/
+export function buildForeignKeyIndexes<TSchema extends DMLSchema>(
+  schema: TSchema
+) {
+  const indexes: EntityIndex<TSchema, QueryCondition>[] = []
+
+  for (const [field, property] of Object.entries(schema)) {
+    const parsed = property.parse(field)
+    const isRelationshipType = "type" in parsed
+    const isBelongsTo = isRelationshipType && parsed.type === "belongsTo"
+
+    if (!isBelongsTo) {
+      continue
+    }
```

Comment:
```suggestion
    if (!BelongsTo.isBelongsTo(property)) {
      continue
    }
```

Reply:
I ve seen multiple places where we are checking the parsed type, but all the relation modifier have a static util to check if it is one of the relation. If some where we need to know if it is a relation of any kind we can add the same static util in the BaseRelation class like this:
```ts
const IsRelationship = Symbol.for("isRelationship")

/**
 * The BaseRelationship encapsulates the repetitive parts of defining
 * a relationship
 */
export abstract class BaseRelationship<T> implements RelationshipType<T> {
  [IsRelationship]: true = true

  static isRelationship<T>(relation: any): relation is BaseRelationship<T> {
    return !!relation?.[IsRelationship]
  }
```